test/tracecontext/tracecontext.test.js

```diff
@@ -0,0 +1,126 @@
+'use strict'
+const agent = require('../..').start({
+  serviceName: 'test-tracecontext',
+  breakdownMetrics: false,
+  captureExceptions: false,
+  metricsInterval: 0,
+  centralConfig: false,
+  cloudProvider: 'none',
+  spanStackTraceMinDuration: 0 // Always have span stacktraces.
+})
+
+const tape = require('tape')
+const TraceContext = require('../../lib/tracecontext')
+const TraceState = require('../../lib/tracecontext/tracestate')
+const TraceParent = require('traceparent')
+
+tape.test('propagateTraceContextHeaders tests', function (suite) {
+  suite.test('Span test', function (t) {
+    const traceParentString = '00-d3ced7e155ca7d275540a77e6ed5f931-ee2afc1f78c2cfa6-01'
+    const traceStateString = 'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de,34@ree=xxxy'
+
+    const transaction = agent.startTransaction('test-transaction', null, {
+      childOf: traceParentString,
+      tracestate: traceStateString
+    })
+
+    const span = transaction.startSpan('test-span')
+    t.true(!span._hasPropagatedTraceContext)
+    const newHeaders = {}
+    span.propagateTraceContextHeaders(newHeaders, function (carrier, name, value) {
+      carrier[name] = value
+    })
+
+    span.end()
+    transaction.end()
+
+    t.equals(span._context.traceparent.toString(), newHeaders.traceparent)
+    t.equals(traceStateString, newHeaders.tracestate)
+    t.equals(span._context.traceparent.toString(), newHeaders['elastic-apm-traceparent'])
+    t.true(span._hasPropagatedTraceContext)
+    t.end()
+  })
+
+  suite.test('TraceContext test', function (t) {
+    const traceParentString = '00-d3ced7e155ca7d275540a77e6ed5f931-ee2afc1f78c2cfa6-01'
+    const traceStateString = 'foo=34f067aa0ba902b7,bar=0.25,es=a:b;cee:de,34@ree=xxxy'
+
+    const fromContext = new TraceContext(
+      TraceParent.fromString(traceParentString),
+      TraceState.fromStringFormatString(traceStateString)
+    )
+    const context = new TraceContext()
+    const headers = {}
+
+    const newHeaders = Object.assign({}, headers)
+    context.propagateTraceContextHeaders(
+      newHeaders,
+      fromContext,
+      function (carrier, name, value) {
+        if (!value) {
+          return
+        }
```

Comment:
context.propagateTraceContextHeaders should always make sure that the value is not empty
```suggestion
```

Reply:
Seems harmless in a test, but also harmless enough to fix.  I've done in in a separate commit -- the setter callbacks no longer have this guard.